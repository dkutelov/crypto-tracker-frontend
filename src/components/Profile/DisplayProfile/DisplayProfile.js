import { Link } from 'react-router-dom';
import styles from './DisplayProfile.module.css';

const DisplayProfile = ({ profile }) => {
  const { _id, avatarUrl, email, firstName, lastName } = profile;
  return (
    <div>
      <div className={styles.avatarWrapper}>
        <h3 className={styles.avatarHeading}>Avatar</h3>
        <img src={avatarUrl} alt='my avatar' className={styles.avatar} />
      </div>
      <div className={styles.infoWrapper}>
        <p className={styles.infoLabel}>First Name</p>
        <p className={styles.infoData}>{firstName}</p>
        <p className={styles.infoLabel}>Last Name</p>
        <p className={styles.infoData}>{lastName}</p>
        <p className={styles.infoLabel}>Email</p>
        <p className={styles.infoData}>{email}</p>

        <div className={styles.editLinkWrapper}>
          <Link to={`/profile/edit`} className={styles.editLink}>
            Edit
          </Link>
        </div>
      </div>
    </div>
  );
};

export default DisplayProfile;
