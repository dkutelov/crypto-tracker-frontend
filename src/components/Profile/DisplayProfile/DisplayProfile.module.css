.avatarWrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.avatar {
  width: 250px;
  height: 250px;
  border-radius: 50%;
}

.avatarHeading {
  text-align: center;
  font-weight: normal;
}

.infoWrapper {
  display: grid;
  grid-template-columns: 1fr 1fr;
  column-gap: 1em;
  margin: 1.5em 0;
}

.infoLabel {
  background-color: #fbe300;
  padding: 0.5em 1em;
}

.infoData {
  padding: 0.5em 1em;
}

.editLinkWrapper {
  grid-column-start: 1;
  grid-column-end: span 2;
  padding-top: 2em;
  display: flex;
  justify-content: center;
}

.editLink {
  display: block;
  width: 60%;
  color: white;
  background-color: #27be94;
  padding: 10px 20px;
  border-radius: 5px;
  text-transform: uppercase;
  text-decoration: none;
  text-align: center;
  transition: all 0.3s;
  border: none;
}

.editLink:hover {
  background-color: #58e7bd;
}
