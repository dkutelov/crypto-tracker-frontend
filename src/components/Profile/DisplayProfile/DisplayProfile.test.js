import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import DisplayProfile from './DisplayProfile';

describe('Display Profile Component', () => {
  const profile = {
    avatarUrl: 'https://www.test.com/avatar.jpg',
    email: 'text@test.com',
    firstName: 'John',
    lastName: 'Doe',
  };

  it('Avatar should have proper src', () => {
    render(
      <BrowserRouter>
        <DisplayProfile profile={profile} />
      </BrowserRouter>
    );

    expect(document.querySelector('img').src).toBe(
      'https://www.test.com/avatar.jpg'
    );
  });

  it('Should match snapshot', () => {
    const component = renderer.create(
      <BrowserRouter>
        <DisplayProfile profile={profile} />
      </BrowserRouter>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
