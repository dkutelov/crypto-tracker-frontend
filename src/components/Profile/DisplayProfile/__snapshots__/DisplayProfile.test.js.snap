// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Profile Component Should match snapshot 1`] = `
<div>
  <div
    className="avatarWrapper"
  >
    <h3
      className="avatarHeading"
    >
      Avatar
    </h3>
    <img
      alt="my avatar"
      className="avatar"
      src="https://www.test.com/avatar.jpg"
    />
  </div>
  <div
    className="infoWrapper"
  >
    <p
      className="infoLabel"
    >
      First Name
    </p>
    <p
      className="infoData"
    >
      John
    </p>
    <p
      className="infoLabel"
    >
      Last Name
    </p>
    <p
      className="infoData"
    >
      Doe
    </p>
    <p
      className="infoLabel"
    >
      Email
    </p>
    <p
      className="infoData"
    >
      text@test.com
    </p>
    <div
      className="editLinkWrapper"
    >
      <a
        className="editLink"
        href="/profile/edit"
        onClick={[Function]}
      >
        Edit
      </a>
    </div>
  </div>
</div>
`;
